.github/workflows/build-and-publish.yml

```
@@ -0,0 +1,174 @@
+          # Extract clean version without prerelease suffix for AssemblyVersion/FileVersion
+          CLEAN_VERSION=$(echo "$VERSION" | sed 's/-.*$//')
+          echo "clean_version=$CLEAN_VERSION" >> $GITHUB_OUTPUT
+          echo "Clean version for assemblies: $CLEAN_VERSION"
+          
+      - name: Update version in project file
+        run: |
+          VERSION=${{ steps.version.outputs.version }}
+          CLEAN_VERSION=${{ steps.version.outputs.clean_version }}
+          echo "Updating project file with version: $VERSION"
+          echo "Using clean version for assemblies: $CLEAN_VERSION"
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)